ServiceApp/src/org/droidplanner/services/android/core/gcs/follow/Follow.java

```diff
@@ -103,11 +123,40 @@ public boolean isEnabled() {
         return state == FollowStates.FOLLOW_RUNNING || state == FollowStates.FOLLOW_START;
     }
 
+    public FollowLocationSource getLocationSource() {
+        return mLocationSource;
+    }
+
+    public void setLocationSource(FollowLocationSource source) {
+        if(mLocationSource != source) {
+            switch(mLocationSource) {
```

Comment:
@kellyschrock this could be confusing. let's have the `switch` on `source` instead, and have explicit `case` block for the `INTERNAL` and `EXTERNAL` location source. 
The `default` block should match the `INTERNAL` case block.